### PR TITLE
Feature/emcpy 76

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**

### CODE REVIEW

**Comment by EdwardSafford-NOAA** [comment]
Added CreateFigure class to create_plots.py.  This is an extension of the matplotlib.figure.Figure class and provides an add_legend() method which will produce legend entries for all sub-figures including those on shared X and/or Y axes.

File: examples/line_plots/multi_line_plot.py [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
 lp3.color = 'tab:red'
 lp3.label = 'line 3'
+lp3.use_shared_ay()
 
-# Add line plot objects to list
-plt_list = [lp1, lp2, lp3]
-
-# Create Plot and draw data
 myplt = CreatePlot()
+plt_list = [lp1, lp2, lp3]
```
Comment (CoryMartin-NOAA) [inline examples/line_plots/multi_line_plot.py:38]: Does the order in which this list is created matter?

File: src/emcpy/plots/create_plots.py [hunk 2]
```diff
@@ -21,8 +22,7 @@ def __init__(self, figsize):
         Args:
             figsize : (tuple) Figure dimension size
         """
-
-        self.fig = plt.figure(figsize=figsize)
+        self.fig = plt.figure(figsize=figsize, FigureClass=CreateFigure)
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/create_plots.py:25]: This is nice!

**Review by CoryMartin-NOAA — DISMISSED** [review]
The figure class inheritance to matplotlib is pretty cool!

File: src/emcpy/plots/create_plots.py [hunk 3]
```diff
@@ -800,3 +799,45 @@ def _add_rivers(self):
...
+        Args:
+            plotobj : (CreatePlot) object for bbox transformation.
+            loc : (str; default='upper left') location of legend box.
+        """
+        coords = self._getcoords(loc)
+        self.legend(loc=loc, bbox_to_anchor=coords, bbox_transform=plotobj.ax.transAxes)
+
+    def _getcoords(self, loc):
+        """
+        Return the relative coordinate tupal for a given location string.
```
Comment (kevindougherty-noaa) [inline src/emcpy/plots/create_plots.py]: Just a quick typo. Should be 'tuple'

**Review by kevindougherty-noaa — DISMISSED** [review]
Just a minor typo in the doc strings, but otherwise looks good to me.

File: examples/line_plots/multi_line_plot.py (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply examples/line_plots/multi_line_plot.py:38]: No, the order of the pot objects should not have any impact on the plot.  Though now that you ask I wonder if an attempt to plot on the shared_x or shared_y axes before anything is plotted on the default ax axes would result in a problem.  I'll test that tomorrow.  If there's a problem I'll open a new issue for it.

File: src/emcpy/plots/create_plots.py (hunk 3 above)
Comment (EdwardSafford-NOAA) [reply src/emcpy/plots/create_plots.py]: Yes, indeed.

**Comment by EdwardSafford-NOAA** [comment]
I realized this morning that I failed to provide all the available arguments for the CreateFigure.add_legend() method.  AFAIK they are the same as the CreatePlot.add_legend() method, so it shouldn't take much work.  Let me look into that before considering merging this.

File: examples/line_plots/multi_line_plot.py (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply examples/line_plots/multi_line_plot.py:38]: I ran some tests and can confirm the order of the object list does not matter.

**Comment by EdwardSafford-NOAA** [comment]
The interface for the CreateFigure.add_legend() method now has the full set of arguments.  From my perspective this is good to go now.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**